lib/wallet/test/unit/Cardano/Wallet/Address/DerivationSpec.hs

```diff
@@ -222,21 +226,28 @@ prop_roundtripEnumIndexSoft ix =
     (toEnum . fromEnum) ix === ix .&&. (toEnum . fromEnum . getIndex) ix === ix
 
 prop_roundtripXPrv
-    :: (PersistPrivateKey (k 'RootK), Eq (k 'RootK XPrv), Show (k 'RootK XPrv))
-    => (k 'RootK XPrv, PassphraseHash)
+    :: forall k
+     . (Eq (k 'RootK XPrv), Show (k 'RootK XPrv))
+    => KeyFlavorS k
+    -> (k 'RootK XPrv, PassphraseHash)
     -> Property
-prop_roundtripXPrv xpriv = do
-    let xpriv' = (unsafeDeserializeXPrv . serializeXPrv) xpriv
+prop_roundtripXPrv keyF xpriv = do
+    let xpriv' =
+            ( unsafeDeserializeXPrv keyF
+                . serializeXPrv keyF
+            )
+                xpriv
     xpriv' === xpriv
```

Comment:
Hmm. Perhaps this could be written more concisely:

```suggestion
prop_roundtripXPrv keyF xpriv =
    xpriv' === xpriv
  where
    xpriv' = unsafeDeserializeXPrv keyF . serializeXPrv keyF $ xpriv
```